projector-awt/src/main/kotlin/org/jetbrains/projector/awt/PWindow.kt

```diff
@@ -179,6 +179,12 @@ class PWindow private constructor(val target: Component, private val isAgent: Bo
     }
   }
 
+  val renderingScale: Float
+    get() {
+      // Assuming uniform x/y scale as is the case for most HiDPI setups
+      return graphics.transform.scaleX.toFloat()
```

Comment:
Hmm, does graphics always have a transformation that represents HiDPI scaling? Looks like no, but maybe I don't see something...

Reply:
It is the case on JBR (IIRC), and it's also the case for non-agent mode: see `PWindow#updateGraphics`. It updates the transform to that of `PGraphicsConfiguration`, which, in turn, replicates JBR behaviour.